packages/vuetify/src/components/VInfiniteScroll/VInfiniteScroll.tsx

```diff
@@ -0,0 +1,257 @@
+// Styles
+import './VInfiniteScroll.sass'
+
+// Components
+import { VBtn } from '@/components/VBtn'
+import { VProgressCircular } from '@/components/VProgressCircular'
+
+// Composables
+import { makeDimensionProps, useDimension } from '@/composables/dimensions'
+import { useIntersectionObserver } from '@/composables/intersectionObserver'
+import { useLocale } from '@/composables/locale'
+
+// Utilities
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
+import { convertToUnit, defineComponent, useRender } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+
+export type InfiniteScrollSide = 'start' | 'end' | 'both'
+export type InfiniteScrollStatus = 'ok' | 'empty' | 'loading' | 'error'
+
+const VInfiniteScrollIntersect = defineComponent({
+  name: 'VInfiniteScrollIntersect',
+
+  props: {
+    side: {
+      type: String as PropType<InfiniteScrollSide>,
+      required: true,
+    },
+    rootRef: null,
+    rootMargin: String,
+  },
+
+  emits: {
+    intersect: (side: InfiniteScrollSide) => true,
+  },
+
+  setup (props, { emit }) {
+    const { intersectionRef, isIntersecting } = useIntersectionObserver(entries => {
+    }, props.rootMargin ? {
+      root: props.rootRef,
+      rootMargin: props.rootMargin,
+    } : undefined)
+
+    watch(isIntersecting, async val => {
+      if (val) emit('intersect', props.side)
+    })
+
+    useRender(() => (
+      <div class="v-infinite-scroll-intersect" ref={ intersectionRef }>&nbsp;</div>
+    ))
+  },
+})
+
+export const VInfiniteScroll = defineComponent({
```

Comment:
will need to be updated with the new genericComponent fn so we can add slots typing. example https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/components/VBtn/VBtn.tsx#L84